src/main/java/org/jenkinsci/plugins/codesonar/models/ProcedureMetric.java

```diff
@@ -0,0 +1,74 @@
+package org.jenkinsci.plugins.codesonar.models;
+
+import java.util.Objects;
+
+public class ProcedureMetric implements Comparable<ProcedureMetric> {

```

Comment:
The name of this class should probably be `ProcedureJsonRow` instead of `ProcedureMetric` since it implicitly stores data parsed from a procedure search result row.  A procedure row stores multiple metrics, so it doesn't make sense to call this class `ProcedureMetric` if we use it this way.  I suppose that means that we should also move this file to the `json` subpackage.

I think when we discussed this code previously, we said that this `ProcedureMetric` class could (in principle) be used for deserializing XML data.  I think that the result produced by by the JSON and XML endpoints are generally different, so I now think we must keep the types used for deserializing the data seperate.  This seems to be yet another reason to move the class into the `json` subpackage.